tests/test_objects.py

```diff
@@ -577,29 +558,12 @@ def test_make_pod_with_env():
             'automountServiceAccountToken': False,
             "containers": [
                 {
-                    "env": [
-                        {
-                            'name': 'TEST_KEY_1',
-                            'value': 'TEST_VALUE',
-                        },
-                        {
-                            'name': 'TEST_KEY_2',
-                            'valueFrom': {
-                                'secretKeyRef': {
-                                    'name': 'my-k8s-secret',
-                                    'key': 'password',
-                                },
-                            },
-                        },
+                    "envFrom": [
                         {
-                            'name': 'TEST_KEY_3',
-                            'valueFrom': {
-                                'secretKeyRef': {
-                                    'name': 'my-k8s-secret',
-                                    'key': 'password',
-                                },
-                            },
-                        },
```

Comment:
Can you ensure old tests are retained and only update what changed? I believe the output 581-584 is what really change, but not the others, right?

```
                        {
                            'name': 'TEST_KEY_1',
                            'value': 'TEST_VALUE',
                        },
```

Reply:
> Can you ensure old tests are retained and only update what changed?

Yes good point, fixing.

Reply:
I changed the tests, added one more test to spawner to verify 'valueFrom' env. variables.